### PR TITLE
Support for custom http headers

### PR DESCRIPTION
**Pull request by EPNW-Eric**
Our Matomo instance is behind an kubernetes ingress which requires authentication. We do this using custom http headers.
Therefor it would be nice if we are able to send those custom headers with each request to the backend.
This PR adds support for custom http headers.

### CODE REVIEW

File: lib/src/matomo_dispatcher.dart
Comment (TesteurManiak) [inline lib/src/matomo_dispatcher.dart:49]: You can use the spread operator directly inside the map:
```dart
final headers = <String, String>{
  if (!kIsWeb && userAgent != null) userAgentHeaderKeys: userAgent,
  ...events.first.tracker.customHeaders,
};
```

**Review by TesteurManiak — CHANGES_REQUESTED** [review]
The code looks good to me (with minor nitpick on the usage of `addAll`), but can you please also add a unit test for this feature. (cc @Pierre-Monier and @MeixDev if you want to double check)

**Comment by EPNW-Eric** [comment]
I've made the requested code change and also added a little test case.
Although not part of the test case, I inspected some requests as seen by our backend and they contained the headers.

File: test/src/matomo_test.dart
Comment (TesteurManiak) [inline test/src/matomo_test.dart:53]: Is this in the "it should be able to initialize" test?
I think it would be better to have a separated specific test.

**Review by TesteurManiak — CHANGES_REQUESTED** [review]

**Comment by TesteurManiak** [comment]
According to the test workflow, your modifications seems to break the tests in `matomo_dispatcher_test.dart`, it should be fixed before merging. (if you need any help to understand the testing practices do not hesitate to ask)

File: test/src/matomo_test.dart
Comment (Pierre-Monier) [reply test/src/matomo_test.dart:53]: Yes, and test that it works with and without custom header

**Comment by EPNW-Eric** [comment]
Unfortunately I'm not familiar with the `mocktail` unit test system and am also not sure what an appropriate test for that feature would be. To put it simple: I've no idea whats going on there.

Futhermore I've looked into the results of the test workflow and it seems that at least 3 of the 4 failed tests are related to `null` errors, which is strange, since the object in question (the header map) is not nullable. Maybe it's related to the mocking system?

As a suggestion, would it be feasible if I revert all test-related changes and change the target branch for the PR to `dev` so that someone more experienced could write a test? I think that would be much faster... If not, I would be glad for any suggestions/help about writing a test case.

**Comment by TesteurManiak** [comment]
> As a suggestion, would it be feasible if I revert all test-related changes and change the target branch for the PR to dev so that someone more experienced could write a test? I think that would be much faster... If not, I would be glad for any suggestions/help about writing a test case.

Sure! That's probably the best thing to do. The failing tests are probably related to an undefined value in a mocked object so it's best that me or another maintainer take a look at it before merging it on the main branch.
I might also write a contribution guide to explain how tests are made on the project as it is important for us that the package doesn't have regression after a contribution.

**Review by TesteurManiak — APPROVED** [review]

**Comment by EPNW-Eric** [comment]
Thank you for your help & time😃!

**Pull request merged**
